vhdl_parser/src/tokenstream.rs

```diff
@@ -127,6 +137,46 @@ impl TokenStream {
     }
 }
 
+pub trait Recover<T> {
+    fn or_recover_until(
+        self,
+        stream: &mut TokenStream,
+        msgs: &mut MessageHandler,
+        cond: fn(&Kind) -> bool,
+    ) -> ParseResult<T>;
+
+    fn log(self, msgs: &mut MessageHandler);
+}
+
+impl<T: std::fmt::Debug> Recover<T> for ParseResult<T> {
+    fn or_recover_until(
+        self,
+        stream: &mut TokenStream,
+        msgs: &mut MessageHandler,
+        cond: fn(&Kind) -> bool,
+    ) -> ParseResult<T> {
+        if self.is_ok() {
+            return self;
+        }
+
+        let res = stream.skip_until(cond);
+        match res {
+            Ok(_) => self,
+            Err(err) => {
+                msgs.push(self.unwrap_err());
```

Comment:
Isnt the message pushed twice? First here and then after the call to or_recover_until above?

Reply:
I think there is a test missing here. The test if implemented would show that the message is pushed twice.